client-metrics-reporter/src/test/java/io/strimzi/kafka/metrics/prometheus/integration/TestConnectMetricsIT.java

```diff
@@ -0,0 +1,275 @@
+            "kafka_connect_kafka_metrics_.*" + CONNECT_ID_PATTERN);
+
+    private static final List<String> SINK_PATTERNS = List.of(
+            "kafka_connect_connector_metrics_.*" + SINK_CONNECTOR_PATTERN,
+            "kafka_connect_connector_task_metrics_.*" + SINK_CONNECTOR_PATTERN,
+            "kafka_connect_sink_task_metrics_.*" + SINK_CONNECTOR_PATTERN,
+            "kafka_connect_task_error_metrics_.*" + SINK_CONNECTOR_PATTERN,
+            "kafka_connect_connect_worker_metrics_connector_count 1.0",
+            "kafka_consumer_app_info_.*" + SINK_CONSUMER_ID,
+            "kafka_consumer_kafka_metrics_.*" + SINK_CONSUMER_ID,
+            "kafka_consumer_consumer_metrics_.*" + SINK_CONSUMER_ID,
+            "kafka_consumer_consumer_node_metrics_.*" + SINK_CONSUMER_ID,
+            "kafka_consumer_consumer_coordinator_metrics_.*" + SINK_CONSUMER_ID,
+            "kafka_consumer_consumer_fetch_manager_metrics_.*" + SINK_CONSUMER_ID);
+
+    private static final List<String> SOURCE_PATTERNS = List.of(
+            "kafka_connect_connector_metrics_.*" + SOURCE_CONNECTOR_PATTERN,
+            "kafka_connect_connector_task_metrics_.*" + SOURCE_CONNECTOR_PATTERN,
+            "kafka_connect_source_task_metrics_.*" + SOURCE_CONNECTOR_PATTERN,
+            "kafka_connect_task_error_metrics_.*" + SOURCE_CONNECTOR_PATTERN,
+            "kafka_connect_connect_worker_metrics_connector_count 2.0",
+            "kafka_producer_app_info_.*" + SOURCE_PRODUCER_ID,
+            "kafka_producer_kafka_metrics_.*" + SOURCE_PRODUCER_ID,
+            "kafka_producer_producer_metrics_.*" + SOURCE_PRODUCER_ID,
+            "kafka_producer_producer_node_metrics_.*" + SOURCE_PRODUCER_ID,
+            "kafka_producer_producer_topic_metrics_.*" + SOURCE_PRODUCER_ID);
+
+    private StrimziKafkaCluster kafka;
+    private StrimziConnectCluster connect;
+
+    @BeforeEach
+    public void setUp() {
+        kafka = new StrimziKafkaCluster.StrimziKafkaClusterBuilder()
+                .withNumberOfBrokers(1)
+                .withSharedNetwork()
+                .build();
+        kafka.start();
+
+        try (Admin admin = Admin.create(Map.of(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers()))) {
+            admin.createTopics(List.of(new NewTopic(TOPIC, 1, (short) -1)));
+        }
+        try (KafkaProducer<String, String> producer = new KafkaProducer<>(Map.of(
+                ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers(),
+                ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName(),
+                ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName()
+        ))) {
+            for (int i = 0; i < 5; i++) {
+                producer.send(new ProducerRecord<>(TOPIC, "record" + i));
+            }
+        }
+    }
+
+    @AfterEach
+    public void tearDown() {
+        connect.stop();
+        kafka.stop();
```

Comment:
Add null check here :) and if it's `!= null` then invoke stop() to prevent potential NPE.